File: quasar/Quasar.DataTransfer/Produtos/Responses/ProdutoResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Quasar.DataTransfer.Categorias.Responses;
using Quasar.DataTransfer.Fornecedores.Responses;

namespace Quasar.DataTransfer.Produtos.Responses
{
    public class ProdutoResponse
    {
        public int Codigo { get; set; }
        public string Descricao { get; set; }
        public string Nome { get; set; }
        public decimal Valor { get; set; }
        public string Imagem { get; set; }
        public FornecedorResponse Fornecedor { get; set; }
        public CategoriaResponse Categoria { get; set; }
        public EspecificacaoResponse Especificacao { get; set; }

    }
}

File: quasar-front/src/app/shared/models/responses/produto.response.ts
import { CategoriaResponse } from "./categoria.response";
import { EspecificacaoResponse } from "./especificacao.response";
import { FornecedorResponse } from "./fornecedor.response";

export class ProdutoResponse {

  public codigo: number;
  public descricao: string;
  public nome: string;
  public valor: number;
  public imagem: string;
  public fornecedor: FornecedorResponse;
  public categoria: CategoriaResponse;
  public especificacao: EspecificacaoResponse


  constructor(params: Partial<ProdutoResponse>) {
    this.codigo = params.codigo || 0;
    this.descricao = params.descricao || '';
    this.nome = params.nome || '';
    this.valor = params.valor || 0;
    this.imagem = params.imagem || '';
    this.fornecedor = params.fornecedor || new FornecedorResponse({});
    this.categoria = params.categoria || new CategoriaResponse({});
    this.especificacao = params.especificacao || new EspecificacaoResponse({});
  }
}
